lib/OpenQA/Worker/Settings.pm

```diff
@@ -0,0 +1,106 @@
+# Copyright (C) 2019 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+package OpenQA::Worker::Settings;
+use Mojo::Base -base;
+use Mojo::Util 'trim';
+
+use Config::IniFiles;
+
+has 'global_settings';
+has 'webui_hosts';
+has 'webui_host_specific_settings';
+
+sub new {
+    my ($class, $instance_number, $cli_options) = @_;
+    $cli_options //= {};
+
+    my $cfg = Config::IniFiles->new(-file => ($ENV{OPENQA_CONFIG} || '/etc/openqa') . '/workers.ini');
+
+    # read global settings from config
+    my %global_settings;
+    for my $section ('global', $instance_number) {
+        if ($cfg && $cfg->SectionExists($section)) {
+            for my $set ($cfg->Parameters($section)) {
+                $global_settings{uc $set} = trim $cfg->val($section, $set);
+            }
+        }
+    }
+
+    # read global settings from environment variables
+    if (my $log_dir = $ENV{OPENQA_WORKER_LOGDIR}) {
+        $global_settings{LOG_DIR} = $log_dir;
+    }
+
+    # read global settings specified via CLI arguments
+    $global_settings{LOG_LEVEL} = 'debug' if $cli_options->{verbose};
+
+    # determine web UI host
+    my $webui_host = $cli_options->{host} || $global_settings{HOST} || 'localhost';
+    delete $global_settings{HOST};
+
+    # determine web UI host specific settings
+    my %webui_host_specific_settings;
+    my @hosts = split(' ', $webui_host);
+    for my $section (@hosts) {
+        if ($cfg && $cfg->SectionExists($section)) {
+            for my $set ($cfg->Parameters($section)) {
+                $webui_host_specific_settings{$section}->{uc $set} = trim $cfg->val($section, $set);
+            }
+        }
+        else {
+            $webui_host_specific_settings{$section} = {};
+        }
+    }
+
+    # set some environment variables
+    # TODO: This should be sent to the scheduler to be included in the worker's table.
```

Comment:
Don't add disabled code as nobody but you will understand why this is not active. Better leave it out and keep in your local git repository, either in `git stash` or a temporary "WIP"-commit or ticket

Reply:
How is that disabled code. I've just copied the comment from one location to another.

Reply:
sorry, wrong github review template ;) Should be: Use tickets for tracking tasks, not code :)